### PR TITLE
Use nullptr

### PR DESCRIPTION
**Pull request by hexfs**

### CODE REVIEW

**Comment by agaida** [comment]
@tsujan @palinek - would you be so kind and review?

File: panel/config/addplugindialog.cpp [hunk 1]
```diff
@@ -81,6 +81,7 @@ AddPluginDialog::AddPluginDialog(QWidget *parent):
 AddPluginDialog::~AddPluginDialog()
 {
     delete ui;
+    ui = nullptr;
```
Comment (palinek) [inline panel/config/addplugindialog.cpp]: No need for this...as this is a destructor and so all the members get out of scope right after this call...

File: plugin-cpuload/lxqtcpuloadplugin.cpp [hunk 2]
```diff
@@ -48,11 +48,13 @@ LXQtCpuLoadPlugin::LXQtCpuLoadPlugin(const ILXQtPanelPluginStartupInfo &startupI
 LXQtCpuLoadPlugin::~LXQtCpuLoadPlugin()
 {
     delete mWidget;
+    mWidget = nullptr;
 }
 
 QWidget *LXQtCpuLoadPlugin::widget()
 {
     return mWidget;
+    mWidget = nullptr;
```
Comment (palinek) [inline plugin-cpuload/lxqtcpuloadplugin.cpp]: Are you sure here? It seems terribly wrong on first look.

File: panel/config/addplugindialog.cpp (hunk 1 above)
Comment (palinek) [reply panel/config/addplugindialog.cpp]: If you insist in this 'delete x; x = nullptr;` pattern, I would prefer to go more deeply and use smart pointers.

**Comment by hexfs** [comment]
@tsujan  @palinek Could you review again?

File: panel/config/configpanelwidget.cpp [hunk 3]
```diff
@@ -305,6 +305,7 @@ int ConfigPanelWidget::indexForPosition(int screen, ILXQtPanel::Position positio
 ConfigPanelWidget::~ConfigPanelWidget()
 {
     delete ui;
+    ui = nullptr;
```
Comment (palinek) [inline panel/config/configpanelwidget.cpp]: ..still the same pattern as noted before.... and the same applies for next occurrences.

Comment (hexfs) [reply panel/config/configpanelwidget.cpp]: Oh, I forgot to remove that

File: plugin-directorymenu/directorymenu.cpp [hunk 4]
```diff
@@ -87,7 +86,6 @@ void DirectoryMenu::buildMenu(const QString& path)
     if(mMenu)
     {
         delete mMenu;
-        mMenu = 0;
```
Comment (palinek) [inline plugin-directorymenu/directorymenu.cpp]: This removal is also wrong and will cause SEGFAULTS IMO.

Comment (hexfs) [reply plugin-directorymenu/directorymenu.cpp]: `mMenu = nullptr;` or `mMenu = 0;`?

Comment (palinek) [reply plugin-directorymenu/directorymenu.cpp]: Oh.. I forgot to look further... the `mMenu` is assigned right after....

**Comment by tsujan** [comment]
IMO, it's GTM now.

@hexfs, please keep up the good work in other sources too!

**Comment by agaida** [comment]
palinek - GTM?

File: plugin-directorymenu/directorymenu.cpp [hunk 5]
```diff
@@ -61,7 +61,6 @@ DirectoryMenu::~DirectoryMenu()
...
```
Comment (palinek) [inline plugin-directorymenu/directorymenu.cpp]: Please, remove the whole `if(mMenu)` condition... just `delete mMenu;`

File: plugin-directorymenu/directorymenu.cpp (hunk 4 above)
Comment (palinek) [reply plugin-directorymenu/directorymenu.cpp]: The same here... just `delete mMenu;`

File: panel/lxqtpanel.cpp [hunk 6]
```diff
@@ -352,6 +352,7 @@ LXQtPanel::~LXQtPanel()
 {
     mLayout->setEnabled(false);
     delete mAnimation;
+    mAnimation = nullptr;
```
Comment (palinek) [inline panel/lxqtpanel.cpp]: This is still unneeded..

**Comment by hexfs** [comment]
@tsujan @palinek

**Comment by tsujan** [comment]
@hexfs To me, it was good and it's also good now. I prefer to do nullity check, albeit redundantly, but @palinek is the expert in the panel code.

**Comment by hexfs** [comment]
@palinek If It is ready to go I can squash the commits.

**Comment by palinek** [comment]
GTM

**Comment by hexfs** [comment]
@palinek

**Pull request merged**